datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1534,44 +1545,50 @@ impl TableSource for LogicalTableSource {
 }
 
 /// Create a [`LogicalPlan::Unnest`] plan
-pub fn unnest(input: LogicalPlan, column: Column) -> Result<LogicalPlan> {
-    unnest_with_options(input, column, UnnestOptions::new())
+pub fn unnest(input: LogicalPlan, columns: Vec<Column>) -> Result<LogicalPlan> {
+    unnest_with_options(input, columns, UnnestOptions::new())
 }
 
 /// Create a [`LogicalPlan::Unnest`] plan with options
 pub fn unnest_with_options(
     input: LogicalPlan,
-    column: Column,
+    columns: Vec<Column>,
     options: UnnestOptions,
 ) -> Result<LogicalPlan> {
-    let (unnest_qualifier, unnest_field) =
-        input.schema().qualified_field_from_column(&column)?;
-
     // Extract the type of the nested field in the list.
-    let unnested_field = match unnest_field.data_type() {
-        DataType::List(field)
-        | DataType::FixedSizeList(field, _)
-        | DataType::LargeList(field) => Arc::new(Field::new(
-            unnest_field.name(),
-            field.data_type().clone(),
-            unnest_field.is_nullable(),
-        )),
-        _ => {
-            // If the unnest field is not a list type return the input plan.
-            return Ok(input);
-        }
-    };
+    let mut unnested_fields: HashMap<usize, _> = HashMap::with_capacity(columns.len());
+    // Add qualifiers to the columns.
+    let mut qualified_columns = Vec::with_capacity(columns.len());
+    for c in &columns {
+        let index = input.schema().index_of_column(c)?;
+        let (unnest_qualifier, unnest_field) = input.schema().qualified_field(index);
+        let unnested_field = match unnest_field.data_type() {
+            DataType::List(field)
+            | DataType::FixedSizeList(field, _)
+            | DataType::LargeList(field) => Arc::new(Field::new(
+                unnest_field.name(),
+                field.data_type().clone(),
+                // Unnesting may produce NULLs even if the list is not null.
+                // For example: unnset([1], []) -> 1, null
```

Comment:
Sadly I don't think there is currently a way to create a list with nullable as false.

Reply:
I think so too. `true` should be equivalent to the previous `unnest_field.is_nullable()`. 
I planned to write some tests related to DataFrame, which could potentially implement it. But the changes in this PR are larger than I expected, so those tests have been delayed.

Reply:
> I planned to write some tests related to DataFrame, which could potentially implement it. But the changes in this PR are larger than I expected, so those tests have been delayed.

We can do it in another PR!

Reply:
> We can do it in another PR!
👌